Source/Entities/EntityBaseCore.generated.cst

```diff
@@ -397,7 +397,7 @@ namespace <%=NameSpace%>
                 <% if(ValidationType == MoM.Templates.ValidationType.EntLib) { %>
                 Validate(e.PropertyName);
                 <% } else { %>
-                 ValidationRules.ValidateRules(e.PropertyName);
+                 ValidationRules.ValidateRules(e.PropertyName, this);

```

Comment:
Is it intentional that this change was not also made in the OnPropertyChanging method?

Reply:
I don't see where this would go in OnPropertyChanging?  Unless I'm missing something?  The comment specifically says not to check ValidationRules.ValidateRules in OnPropertyChanging?

Reply:
you are right, good call.  I've pulled the code down, and started running it through my tests, so far it is looking good.  Can't wait to get it all merged in.

Reply:
It has made a big difference for us.  But then again we have some ridiculously large tables (column-wise).

Reply:
I'm curious, were you just seeing some really high memory usage on your web servers?  What issue was this targeting?

Reply:
Yes - So we would see large memory spikes in our application (fairly short lived because we dispose effectively).  So when we looked into it we would see millions of instances of exactly the same class taking up 200-500MB of RAM.  For just 1 single Get from a service.  

As an example from one of my debugging sessions....

- We found that Entities.ValidationRuleHandler had 1.48 million instances
- Each instance uses 392 bytes
- 392 bytes * 1.48m = 580MB

All just to return about 5000 rows of data.   Using basic maths, this tells you that the table has approx 296 columns.  A lot I know... but was much easier to fix this memory issue than the table design!

Reply:
Good to know.  I'm pretty excited to have an improvement like this going into the code base.  We may not all have the issue quite to the level you do, but we'll all benefit from your solution.

Reply:
Yeh - this should provide a general decrease in memory requirements.  Some people will benefit more than others, but there should be improvement across the board.

I also suppose there will be a perf improvement too, but I haven't measured it.  In my case, there must be a benefit to not creating 1.48 million objects...